### PR TITLE
Setup automatic releasing

### PR DESCRIPTION
**Pull request by Almouro**

### CODE REVIEW

**Comment by Almouro** [comment]
Setup automatic release with Travis and Semantic release each time a PR is merged on master.
Versioning follows these rules: https://github.com/semantic-release/semantic-release#default-commit-message-format
Forces us to follow angular style commit messages.

We could even add commitizen like [this](https://github.com/bamlab/devicelab-bot/commit/68fd8cbd1e57a2f1ad4e54a217b8b9de21379bfa) to make it easier to write messages.

File: package.json [hunk 1]
```diff
@@ -1,10 +1,12 @@
 {
   "name": "generator-rn-toolbox",
-  "version": "2.0.6",
```
Comment (yleflour) [inline package.json]: Is it normal?

File: .travis.yml [hunk 2]
```diff
@@ -14,3 +14,11 @@ install:
   - echo y | yo rn-toolbox:eslint
   - echo y | yo rn-toolbox:jest
   - echo GeneratorTest | yo rn-toolbox:base --force
+  - cd ..
+before_script:
+  - npm prune
```
Comment (yleflour) [inline .travis.yml]: If yarn.lock is found yarn is available in travis. Travis then runs "yarn" which prunes dependencies automatically.

File: .travis.yml [hunk 3]
```diff
@@ -14,3 +14,11 @@ install:
...
+after_success:
+  - npm run semantic-release
```
Comment (yleflour) [inline .travis.yml]: yarn semantic-release

File: package.json (hunk 1 above)
Comment (Almouro) [inline package.json]: yes, `semantic-release-cli setup` does this by default.
`semantic-release` relies on published version in npm anyway

File: .travis.yml (hunk 2 above)
Comment (Almouro) [inline .travis.yml]: i got `yarn not found`. I kinda wanna get this money now and setup a proper testing CI later :)

File: .travis.yml (hunk 3 above)
Comment (Almouro) [inline .travis.yml]: same here

**Pull request merged**